src/serlio/prtMaterial/ArnoldMaterialNode.cpp

```diff
@@ -0,0 +1,575 @@
+	}
+	else {
+		size_t tmpSize = buf.size();
+		sb.setVar(L"$mapNode", shadingGroupName + L"_roughness_map");
+		sb.setVar(L"$mapFile", prt::StringUtils::toUTF16FromOSNarrow(matInfo.roughnessMap.c_str(), buf.data(), &tmpSize));
+		sb.createTexture(L"$mapNode");
+		sb.setAttr(L"($mapNode + \".fileTextureName\")", L"$mapFile");
+		sb.setVar(L"$uvTrafoNode", shadingGroupName + L"_roughness_map_trafo");
+		sb.createShader(L"aiUvTransform", L"$uvTrafoNode");
+		sb.setAttr(L"($uvTrafoNode + \".uvset\")", L"\"roughnessMap\"");
+		sb.setAttr(L"($uvTrafoNode + \".pivotFrame\")", 0.0, 0.0);
+		sb.setAttr(L"($uvTrafoNode + \".scaleFrame\")", 1.0 / matInfo.roughnessmapTrafo.su(), 1.0 / matInfo.roughnessmapTrafo.sv());
+		sb.setAttr(L"($uvTrafoNode + \".translateFrame\")", -matInfo.roughnessmapTrafo.tu() / matInfo.roughnessmapTrafo.su(), -matInfo.roughnessmapTrafo.tv() / matInfo.roughnessmapTrafo.sv());
+		if (matInfo.roughnessmapTrafo.rw() != 0.0) {
+			LOG_WRN << "rotation (material.map.rw) is not yet supported\n";
+		}
+		// TODO: GlTF defines the roughness to be in the green channel of an RGB texture, do we also need to support single channel textures?
+		sb.connectAttr(L"($mapNode + \".outColorG\")", L"($uvTrafoNode + \".passthroughR\")");
+		sb.connectAttr(L"($uvTrafoNode + \".outColorR\")", L"($roughnessMapBlendNode + \".input2R\")");
+	}
+
+	// metallic/metallic map multiply node
+	sb.setVar(L"$metallicMapBlendNode", shadingGroupName + L"_metallic_map_blend");
+	sb.createShader(L"aiMultiply", L"$metallicMapBlendNode");
+	sb.connectAttr(L"($metallicMapBlendNode + \".outColorR\")", L"($shaderNode + \".metalness\")");
+
+	// metallic
+	sb.setAttr(L"($metallicMapBlendNode + \".input1R\")", matInfo.metallic);
+
+	// metallic map
+	if (matInfo.metallicMap.empty()) {
+		sb.setAttr(L"($metallicMapBlendNode + \".input2R\")", 1.0);
+	}
+	else {
+		size_t tmpSize = buf.size();
+		sb.setVar(L"$mapNode", shadingGroupName + L"_metallic_map");
+		sb.setVar(L"$mapFile", prt::StringUtils::toUTF16FromOSNarrow(matInfo.metallicMap.c_str(), buf.data(), &tmpSize));
+		sb.createTexture(L"$mapNode");
+		sb.setAttr(L"($mapNode + \".fileTextureName\")", L"$mapFile");
+		sb.setVar(L"$uvTrafoNode", shadingGroupName + L"_metallic_map_trafo");
+		sb.createShader(L"aiUvTransform", L"$uvTrafoNode");
+		sb.setAttr(L"($uvTrafoNode + \".uvset\")", L"\"metallicMap\"");
+		sb.setAttr(L"($uvTrafoNode + \".pivotFrame\")", 0.0, 0.0);
+		sb.setAttr(L"($uvTrafoNode + \".scaleFrame\")", 1.0 / matInfo.metallicmapTrafo.su(), 1.0 / matInfo.metallicmapTrafo.sv());
+		sb.setAttr(L"($uvTrafoNode + \".translateFrame\")", -matInfo.metallicmapTrafo.tu() / matInfo.metallicmapTrafo.su(), -matInfo.metallicmapTrafo.tv() / matInfo.metallicmapTrafo.sv());
+		if (matInfo.metallicmapTrafo.rw() != 0.0) {
+			LOG_WRN << "rotation (material.map.rw) is not yet supported\n";
+		}
+		// TODO: GlTF defines the metalness to be in the blue channel of an RGB texture, do we also need to support single channel textures?

```

Comment:
CityEngine has a definition of where metallic or roughness are expectd in terms of maps. its not yet uptodate in the online CGA ref. here's the excerpt from the offline CGA ref:

> The bump map is interpreted as luminance map (CCIR 601).
> The normal map is interpreted as normals in tangent space encoded as rgb colors.
> The roughness map only uses the green channel.
> The metallic map only uses the blue channel.
> The occlusion map only uses the red channel.

Reply:
converted the TODOs to comments and also fixed handling of the bump map, which was not interpreted as luminance map